x/config/config.go

```diff
@@ -171,3 +171,27 @@ func NewPacketListener(transportConfig string) (transport.PacketListener, error)
 		return nil, fmt.Errorf("config scheme '%v' is not supported", url.Scheme)
 	}
 }
+
+func SanitizeConfig(transportConfig string) (string, error) {
+	// Do nothing if the config is empty
+	if transportConfig == "" {
+		return "", nil
+	}
+	// Split the string into parts
+	parts := strings.Split(transportConfig, "|")
```

Comment:
I'm afraid this is not safe. Some new transport may have sensitive parameters.
We need to explicitly list the configs that are safe to output, and output "$scheme:[REDACTED]" for those that aren't (it's safe to output the scheme).